src/test/kotlin/uk/gov/justice/hmpps/offenderevents/service/OffenderEventStoreTest.kt

```diff
@@ -83,5 +83,19 @@ class OffenderEventStoreTest {
         .extracting<String>(DisplayMessage::eventType).containsExactlyInAnyOrder("1")
   }
 
-  private fun aMessage() = Message("ANY_MESSAGE", "ANY_MESSAGE_ID", MessageAttributes(EventType("ANY_EVENT_TYPE")))
+  @Test
+  fun `Nested arrays can be deserialized`() {
+    offenderEventStore.handleMessage(aNestedArrayMessage())
+
+    val stored = offenderEventStore.getPageOfMessages(listOf(), listOf(), "", 1)
+    assertThat(stored.get(0).messageDetails["offenderIdDisplay"]).isEqualTo("G0373GG")
+    assertThat(stored.get(0).messageDetails["offenders"]).contains("1025558")
+  }
+
+  private fun aMessage() = Message("{\"ANY_KEY\": \"ANY_MESSAGE\"}", "ANY_MESSAGE_ID", MessageAttributes(EventType("ANY_EVENT_TYPE")))
+
+  private fun aNestedArrayMessage() = Message(
+      "{\"offenderIdDisplay\":\"G0373GG\",\"offenders\":[{\"offenderId\":1025558,\"bookings\":[{\"offenderBookId\":12678}]}]}",
```

Comment:
An example that previously broke the UI.

Reply:
Sorry, my fault!!

Reply:
Perfectly valid JSON means you no-claims bonus is fully intact :)